processor/src/fast/mod.rs

```diff
@@ -160,22 +164,35 @@ impl FastProcessor {
     /// # Panics
     /// - Panics if the length of `stack_inputs` is greater than `MIN_STACK_DEPTH`.
     pub fn new(stack_inputs: &[Felt]) -> Self {
-        Self::initialize(stack_inputs, false)
+        Self::initialize(stack_inputs, AdviceInputs::default(), false)
+    }
+
+    /// Creates a new `FastProcessor` instance with the given stack and advice inputs.
+    ///
+    /// The stack inputs are expected to be stored in reverse order. For example, if `stack_inputs =
+    /// [1,2,3]`, then the stack will be initialized as `[3,2,1,0,0,...]`, with `3` being on
+    /// top.
+    ///
+    /// # Panics
+    /// - Panics if the length of `stack_inputs` is greater than `MIN_STACK_DEPTH`.
+    pub fn new_with_advice_inputs(stack_inputs: &[Felt], advice_inputs: AdviceInputs) -> Self {
+        Self::initialize(stack_inputs, advice_inputs, false)
     }
 
-    /// Creates a new `FastProcessor` instance with the given stack inputs, set to debug mode.
+    /// Creates a new `FastProcessor` instance with the given stack and advice inputs,
+    /// set to debug mode.
     ///
     /// The stack inputs are expected to be stored in reverse order. For example, if `stack_inputs =
     /// [1,2,3]`, then the stack will be initialized as `[3,2,1,0,0,...]`, with `3` being on
     /// top.
     ///
     /// # Panics
     /// - Panics if the length of `stack_inputs` is greater than `MIN_STACK_DEPTH`.
-    pub fn new_debug(stack_inputs: &[Felt]) -> Self {
-        Self::initialize(stack_inputs, true)
+    pub fn new_debug(stack_inputs: &[Felt], advice_inputs: AdviceInputs) -> Self {
+        Self::initialize(stack_inputs, advice_inputs, true)
     }
 
```

Comment:
Agree with @plafer 's comment about moving the detailed comments to this function